precovery/frame_db.py

```diff
@@ -626,9 +620,66 @@ def load_csv(
                 data_offset=offset,
                 data_length=length,
             )
-            frames_to_add.append(frame)
 
-        self.idx.add_frames(frames_to_add)
+            healpix_frames.append(frame)
+
+        self.idx.add_frames(healpix_frames)
+
+    @staticmethod
+    def _compute_year_month_str(frame: sourcecatalog.SourceFrame) -> str:
```

Comment:
This is unimportant, but maybe this should live as a staticmethod on `sourcecatalog.SourceFrame`? Since you're not passing in more general types here (datetime, date string, etc.), the internal knowlege about `SourceFrame` feels weirdly coupled. I guess that's true for `add_frames` as well and maybe the whole `sourcecatalog` module is really just a sub of `frame_db` since I don't think we really use it anywhere else?

Reply:
It's a little weird, yeah. I kept it here because its purpose is as a partitioning key for the FrameDB; it's not really related to the `SourceFrame` class directly, which is just a container representing input Frame data.

It's true that `sourcecatalog` is only used when loading data into FrameDB - hence the "source" prefix. I could imagine CLI tools, for example to verify a source data file format or count the number of frames it holds, which would want to use `sourcecatalog` directly.

I'm open to reorganization if you like but making `frame_db` into a whole package with `sourcecatalog` as a module seems unnecessary to me.